src/main/java/telraam/logic/lapper/slapper/Slapper.java

```diff
@@ -0,0 +1,139 @@
+import telraam.database.models.LapSource;
+import telraam.logic.lapper.Lapper;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+// Lapper that only uses a single sql query
+public class Slapper implements Lapper {
+    private final String SOURCE_NAME = "slapper";
+    private final int DEBOUNCE_TIMEOUT = 10;
+    private final ScheduledExecutorService scheduler;
+    private boolean debounceScheduled;
+    private final Logger logger;
+    private final Jdbi jdbi;
+
+    public Slapper(Jdbi jdbi) {
+        this.jdbi = jdbi;
+        this.scheduler = Executors.newScheduledThreadPool(1);
+        this.logger = Logger.getLogger(Slapper.class.getName());
+        this.debounceScheduled = false;
+
+        // Get the lapSourceId, create the source if needed
+        LapSourceDAO lapSourceDAO = jdbi.onDemand(LapSourceDAO.class);
+        if (lapSourceDAO.getByName(SOURCE_NAME).isEmpty()) {
+            lapSourceDAO.insert(new LapSource(SOURCE_NAME));
+        }
+    }
+
+    @Override
+    public void handle(Detection msg) {
+        if (!this.debounceScheduled) {
+            this.debounceScheduled = true;
+            this.scheduler.schedule(() -> {
+                try {
+                    this.calculateLaps();
+                } catch (Exception e) {
+                    logger.severe(e.getMessage());
+                }
+                this.debounceScheduled = false;
+            }, DEBOUNCE_TIMEOUT, TimeUnit.SECONDS);
+        }
+    }
+
+    private void calculateLaps() {
+        logger.info("Slapper: Calculating laps...");
+        this.jdbi.useHandle(handle -> handle.execute(
+                        """
+                                WITH switchovers AS (
+                                	SELECT teamid AS team_id,
+                                		newbatonid,
+                                		timestamp AS start_timestamp,
+                                		COALESCE(
+                                			LEAD(timestamp) OVER (PARTITION BY teamid ORDER BY timestamp),
+                                			timestamp + INTERVAL '1 MONTH'
+                                		) AS next_baton_switch
+                                	FROM batonswitchover
+                                ),
+                                team_detections AS (
+                                	SELECT MIN(station_id) AS station_id,
+                                		MAX(rssi) as rssi,
+                                		date_trunc('second', timestamp) AS timestamp_seconds,
+                                		team_id
+                                	FROM detection d
+                                	LEFT JOIN switchovers s ON d.baton_id = s.newbatonid
+                                		AND d.timestamp BETWEEN s.start_timestamp AND s.next_baton_switch
+                                	WHERE station_id NOT BETWEEN 3 AND 5
```

Comment:
What is special about station_id's between 3 and 5?

Reply:
A new lap is registered if a team goes from 6 -> 1, 7 -> 1 or 7 -> 2.
 SL doesn't look at stations between 3 and 5 so no reason to inflate the table with them